server/statistics/region_collection.go

```diff
@@ -128,6 +140,20 @@ func (r *RegionStatistics) deleteOfflineEntry(deleteIndex RegionStatisticType, r
 	}
 }
 
+// RegionStatsNeedUpdate checks whether the region's status need to be updated
+// due to some special state types.
+func (r *RegionStatistics) RegionStatsNeedUpdate(region *core.RegionInfo) bool {
+	regionID := region.GetID()
+	maxSize := int64(r.storeConfigManager.GetStoreConfig().GetRegionMaxSize())
+	maxKeys := int64(r.storeConfigManager.GetStoreConfig().GetRegionMaxKeys())
+	if r.IsRegionStatsType(regionID, OversizedRegion) != region.IsOversized(maxSize, maxKeys) {
```

Comment:
I wonder `IsRegionStatsType` can be removed.

Reply:
We need to put a read lock inside `IsRegionStatsType` and it's also used in the test. I want to keep it.

Reply:
why cannot directly use `region.IsOversized(maxSize, maxKeys)`?

Reply:
I write this check to make sure the `RegionStatistics` will only be updated when it's necessary:

* The region is not oversized before.
* The region is not oversized anymore.